awscli/customizations/s3/s3handler.py

```diff
@@ -367,169 +381,125 @@ def _enqueue_upload_end_task(self, filename, upload_context):
+            bucket, key = find_bucket_key(file.src)
+            return self._download(manager, bucket, key)
+
+    def _download(self, manager, bucket, key):
+        """
+        Download the specified object and print it to stdout.
 
-    def _enqueue_tasks(self, files):
-        total_files = 0
-        total_parts = 0
-        for filename in files:
-            num_uploads = 1
-            # If uploading stream, it is required to read from the stream
-            # to determine if the stream needs to be multipart uploaded.
-            payload = None
-            if filename.operation_name == 'upload':
-                payload, is_multipart_task = \
-                    self._pull_from_stream(self.multi_threshold)
-            else:
-                # Set the file size for the ``FileInfo`` object since
-                # streams do not use a ``FileGenerator`` that usually
-                # determines the size.
-                filename.set_size_from_s3()
-                is_multipart_task = self._is_multipart_task(filename)
-            if is_multipart_task and not self.params['dryrun']:
-                # If we're in dryrun mode, then we don't need the
-                # real multipart tasks.  We can just use a BasicTask
-                # in the else clause below, which will print out the
-                # fact that it's transferring a file rather than
-                # the specific part tasks required to perform the
-                # transfer.
-                num_uploads = self._enqueue_multipart_tasks(filename, payload)
-            else:
-                task = tasks.BasicTask(
-                    session=self.session, filename=filename,
-                    parameters=self.params,
-                    result_queue=self.result_queue,
-                    payload=payload)
-                self.executor.submit(task)
-            total_files += 1
-            total_parts += num_uploads
-        return total_files, total_parts
+        :type manager: s3transfer.manager.TransferManager
+        :param manager: The transfer manager to use for the download.
+
+        :type bucket: str
+        :param bucket: The bucket to download the object from.
 
-    def _pull_from_stream(self, amount_requested):
+        :type key: str
+        :param key: The name of the key to download.
+
+        :return: A CommandResult representing the download status.
         """
-        This function pulls data from stdin until it hits the amount
-        requested or there is no more left to pull in from stdin.  The
-        function wraps the data into a ``BytesIO`` object that is returned
-        along with a boolean telling whether the amount requested is
-        the amount returned.
+        params = {}
+        RequestParamsMapper.map_get_object_params(params, self.params)
+
+        future = manager.download(
+            fileobj=StdoutBytesWriter(), bucket=bucket,
+            key=key, extra_args=params)
```

Comment:
Cool. Another thing that I was a little worried about is that since we combine head object and get object on the download(), I noticed we only map the get_object_params, but since those map_head_object and map_get_object are essentially the same we should be fine and not be leaving out any arguments.

It may be good to make a note about this though so we do not forget or map both of them anyways.